MM crash when try open "modules" without internet connection (310)

```
10-14 20:52:16.014  3248  3248 I NavController: Ignoring popBackStack to destination com.topjohnwu.magisk:id/modulesFragment as it was not found on the current back stack
10-14 20:52:16.376  3248 12819 D OkHttp  : --> GET https://magisk-modules-repo.github.io/submission/modules.json
10-14 20:52:16.380  3248 12819 D OkHttp  : <-- HTTP FAILED: java.net.UnknownHostException: Unable to resolve host "magisk-modules-repo.github.io": No address associated with hostname
10-14 20:52:16.577  3248  3248 E App     : java.net.UnknownHostException: Unable to resolve host "magisk-modules-repo.github.io": No address associated with hostname
10-14 20:52:16.577  3248  3248 E App     : 	at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:156)
10-14 20:52:16.577  3248  3248 E App     : 	at java.net.Inet6AddressImpl.lookupAllHostAddr(Inet6AddressImpl.java:103)
10-14 20:52:16.577  3248  3248 E App     : 	at java.net.InetAddress.getAllByName(InetAddress.java:1152)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.Dns$1.lookup(Dns.java:40)
10-14 20:52:16.577  3248  3248 E App     : 	at com.topjohnwu.magisk.di.DnsResolver.lookup(NetworkingModule.kt:68)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.connection.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:185)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.connection.RouteSelector.nextProxy(RouteSelector.java:149)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.connection.RouteSelector.next(RouteSelector.java:84)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:215)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.java:225)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:201)
10-14 20:52:16.577  3248  3248 E App     : 	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
10-14 20:52:16.577  3248  3248 E App     : 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
10-14 20:52:16.577  3248  3248 E App     : 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
10-14 20:52:16.577  3248  3248 E App     : 	at java.lang.Thread.run(Thread.java:919)
10-14 20:52:16.577  3248  3248 E App     : Caused by: android.system.GaiException: android_getaddrinfo failed: EAI_NODATA (No address associated with hostname)
10-14 20:52:16.577  3248  3248 E App     : 	at libcore.io.Linux.android_getaddrinfo(Native Method)
10-14 20:52:16.577  3248  3248 E App     : 	at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:74)
10-14 20:52:16.577  3248  3248 E App     : 	at libcore.io.BlockGuardOs.android_getaddrinfo(BlockGuardOs.java:200)
10-14 20:52:16.577  3248  3248 E App     : 	at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:74)
10-14 20:52:16.577  3248  3248 E App     : 	at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:135)
10-14 20:52:16.577  3248  3248 E App     : 	... 30 more
10-14 20:52:16.579  3248  3248 I mwszfy.p.i.u.b: System.exit called, status: 1
```
Full log here: [log.txt](https://github.com/topjohnwu/Magisk/files/5379998/log.txt)

--- a/app/src/main/java/com/topjohnwu/magisk/core/tasks/RepoUpdater.kt
+++ b/app/src/main/java/com/topjohnwu/magisk/core/tasks/RepoUpdater.kt
@@ -20,7 +20,7 @@ class RepoUpdater(
         val cachedMap = HashMap<String, Date>().also { map ->
             repoDB.getModuleStubs().forEach { map[it.id] = Date(it.last_update) }
         }.synchronized()
-        svc.fetchRepoInfo()?.also { info ->
+        svc.fetchRepoInfo()?.let { info ->
             coroutineScope {
                 info.modules.forEach {
                     launch {
@@ -36,7 +36,7 @@ class RepoUpdater(
                     }
                 }
             }
+            repoDB.removeModules(cachedMap.keys)
         }
-        repoDB.removeModules(cachedMap.keys)
     }
 }

--- a/app/src/main/java/com/topjohnwu/magisk/data/repository/NetworkService.kt
+++ b/app/src/main/java/com/topjohnwu/magisk/data/repository/NetworkService.kt
@@ -20,7 +20,7 @@ class NetworkService(
     private val jsd: JSDelivrServices,
     private val api: GithubApiServices
 ) {
-    suspend fun fetchUpdate() = try {
+    suspend fun fetchUpdate() = safe {
         var info = when (Config.updateChannel) {
             DEFAULT_CHANNEL, STABLE_CHANNEL -> fetchStableUpdate()
             BETA_CHANNEL -> fetchBetaUpdate()
@@ -36,12 +36,6 @@ class NetworkService(
         }
         Info.remote = info
         info
-    } catch (e: IOException) {
-        Timber.e(e)
-        null
-    } catch (e: HttpException) {
-        Timber.e(e)
-        null
     }
 
     // UpdateInfo
@@ -69,7 +63,7 @@ class NetworkService(
     private inline fun <T> safe(factory: () -> T): T? {
         return try {
             factory()
-        } catch (e: HttpException) {
+        } catch (e: Exception) {
             Timber.e(e)
             null
         }

--- a/app/src/main/java/com/topjohnwu/magisk/ui/install/InstallViewModel.kt
+++ b/app/src/main/java/com/topjohnwu/magisk/ui/install/InstallViewModel.kt
@@ -18,6 +18,8 @@ import com.topjohnwu.magisk.utils.set
 import com.topjohnwu.superuser.Shell
 import kotlinx.coroutines.launch
 import org.koin.core.get
+import timber.log.Timber
+import java.io.IOException
 import kotlin.math.roundToInt
 
 class InstallViewModel(
@@ -64,7 +66,11 @@ class InstallViewModel(
 
     init {
         viewModelScope.launch {
-            notes = svc.fetchString(Info.remote.magisk.note)
+            try {
+                notes = svc.fetchString(Info.remote.magisk.note)
+            } catch (e: IOException) {
+                Timber.e(e)
+            }
         }
     }
 

